### ISSUE TITLE
EKS: EKS uses AL2_x86_64 AmiType in AWS::EKS::Nodegroup instead of AL2023_x86_64_STANDARD

### ISSUE THREAD

**Issue by markusl** [aws/aws-cdk#32211]
### Describe the bug

EKS: EKS uses AL2_x86_64 AmiType in AWS::EKS::Nodegroup instead of AL2023_x86_64_STANDARD

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Expected Behavior

I would expect the latest AMI to be used instead of the old one. Once a cluster is deployed, a message is shown in AWS console: "New AMI release version is available for this node group."

Terraform module also suggests that "Starting on 1.30, AL2023 is the default AMI type for EKS managed node groups"
https://registry.terraform.io/modules/terraform-aws-modules/eks/aws/20.29.0

### Current Behavior

The following template is generated:
```JSON
  "EksClusterNodegroupDefaultCapacityE6D932C8": {
   "Type": "AWS::EKS::Nodegroup",
   "Properties": {
    "AmiType": "AL2_x86_64",
...

```

### Reproduction Steps

```ts
      const cluster = new eks.Cluster(this, 'EksGpuCluster', {
        vpc: vpc,
        vpcSubnets: [{ subnetType: ec2.SubnetType.PUBLIC }],
        version: eks.KubernetesVersion.V1_31,
        defaultCapacity: 2,
        defaultCapacityInstance: new ec2.InstanceType('m5.xlarge'),
        kubectlLayer: new KubectlV31Layer(this, 'KubectlLayer'),
        outputConfigCommand: true,
        outputClusterName: true,
        authenticationMode: eks.AuthenticationMode.API_AND_CONFIG_MAP,
      });
```

### CDK CLI Version

2.167.1 (build d681b12)

### Node.js Version

20

### OS

Mac

### Language

TypeScript

**Comment by pahud** [comment]
Yes

https://github.com/aws/aws-cdk/blob/a8bc46d80d5f29f511eebc74c8482c1ac2675349/packages/aws-cdk-lib/aws-eks/lib/managed-nodegroup.ts#L488

If props.amiType is not specified, the first element of `possibleAmiTypes` would be picked, which is [AL2_X86_64](https://github.com/aws/aws-cdk/blob/a8bc46d80d5f29f511eebc74c8482c1ac2675349/packages/aws-cdk-lib/aws-eks/lib/managed-nodegroup.ts#L581C20-L581C30).

We'll have to shift the order [here](https://github.com/aws/aws-cdk/blob/a8bc46d80d5f29f511eebc74c8482c1ac2675349/packages/aws-cdk-lib/aws-eks/lib/managed-nodegroup.ts#L580-L582) and this would cause breaking changes and requires a feature flag.

We welcome community PRs.

**Comment by markusl** [comment]
@pahud Yes, I noticed `amiType` for `addNodegroupCapacity`. However, the `defaultCapacity` and `defaultCapacityInstance` properties of `eks.Cluster` do not seem to make it possible to specify AL2023 / Bottlerocket. Is there a workaround for that?

**Comment by pahud** [comment]
@markusl 

I am afraid not. That defaultCapacity is an opinionated design. I would suggest to set it to 0 in the cluster props and 

cluster.addNodegroupCapacity() to create a customized nodegroup capacity instead.

**Comment by markusl** [comment]
@pahud okay, thanks for explaining. What are the benefits of having AL2 instead of AL2023 or Bottlerocket? Should I just ignore the message in the AWS Console which says "New AMI release version is available for this node group."?

**Comment by plumdog** [comment]
I found this issue because I got an email from AWS saying:

> Amazon EKS will end support for EKS optimized AL2 AMIs on November 26, 2025.

I understand I can explicitly set eg `amiType: NodegroupAmiType.AL2023_X86_64_STANDARD` for Nodegroups in my CDK code and this will then migrate them to use AL2023 AMIs instead of AL2 AMIs.

The CDK should default to AL2023 (with suitable management via feature flags).

The CDK could add an option to allow me to say "use whatever AL2023 AMI type works with my instance types" which is currently how it behaves by default, but only for AL2.

**Comment by ckappen** [comment]
We created a new EKS managed node group this morning via AWS CLI without explicitly setting amiType.
Previously, the default was Amazon Linux 2 (AL2_x86_64), but this time the node group was launched with Amazon Linux 2023 (AL2023_x86_64_STANDARD).

This caused unexpected behavior in our cluster since our existing setup relies on AL2.

**Comment by jakeseg** [comment]
AL2_x86_64 will be deprecated soon, and  EKS 1.33 nodes not support AL2. So whats the proper solution?
Use CUSTOM and provide the AMI ID?

**Comment by gudipati** [comment]
As of CDK v2.259.0, you can use the `@aws-cdk/aws-eks:defaultToAL2023` feature flag to default to AL2023 AMI type
